### DISCUSSION TITLE
Provide template for bug report as repository

### DISCUSSION THREAD

**Discussion by Bananeweizen** [checkstyle/checkstyle#13773] [Ideas]
The bug reporting requires people to have a small config, a Java class for reproduction and to execute some command lines. While that is documented nicely in the issue template, it might be even more convinient for people reporting bugs, if there were a separate repository containing an empty `config.xml` and `Bug.java`. Someone who wants to report a bug would be able to fork that repository, modify the config.xml and Bug.java and could immediately push that commit in his fork, as a complete reproduction example. A Github action would then start a custom workflow on the forked repository producing all the output that is currently created via copy/paste (e.g. the cat output and the checkstyle output).

This would reduce the effort for reporters quite a lot, since they would only copy 2 files into their fork and push Everything else would be fully automated instead of just documented (in the template).

**Comment by romani** [comment]
We had an idea to simply make action to get details from issue description and execute as required.  So user only needs to provide chunk of config and java file as text.

But having ideas is good but we are sinking in issues, and nobody who can do this. But we are welcome contribution.

**Comment by Bananeweizen** [comment]
Proof of concept available at https://github.com/Bananeweizen/bug-report. Can be triggered at https://github.com/Bananeweizen/bug-report/actions/workflows/report-bug.yml.

To make it production ready, someone would need to create a mini website for taking the input of Java file and config file, and then trigger the workflow via REST. While you _can_ give these 2 inputs to my workflow already, Github actions only supports single line input fields, and you cannot make a user convert the 2 files to single line input manually.

**Reply by piyush-1234** [reply]
hi, i am new here, looking forword to work on checkstyle's issues, can i know where to join the chat, looks like gitter chat is not opening

**Reply by romani** [reply]
@piyush-1234 , please create separate discussion thread

**Reply by piyush-1234** [reply]
sure
